#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/Floor.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/ThreadMetalPlate.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/Parquetry.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.2731450262637656 0.2999980681626678 0.9139983335466395 1.5669600698501367
  position -0.3910059795201675 -8.655139351598343 5.631292541573186
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
  size 20 20
  appearance Parquetry {
  }
}
DEF skid_steer Robot {
  translation 0.0146476 -0.983017 0.46
  rotation 0 0 1 -2.3561953071795863
  children [
    DEF hcsr2 DistanceSensor {
      translation -0.51 2.73307e-06 0
      rotation 0 0 1 3.14159
      children [
        DEF hcsr2 Solid {
          translation -0.01 0 0
          children [
            DEF hcsr2 Shape {
              appearance PBRAppearance {
                baseColor 0.752941 0.380392 0.796078
                metalness 0
              }
              geometry Capsule {
                height 0.01
                radius 0.01
              }
            }
          ]
          name "hcsr2"
        }
      ]
      name "hcsr2"
    }
    DEF hcsr1 DistanceSensor {
      translation 0.52 0 0
      children [
        Solid {
          translation -0.01 0 0
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.752941 0.380392 0.796078
                metalness 0
              }
              geometry Capsule {
                height 0.01
                radius 0.01
              }
            }
          ]
        }
      ]
    }
    DEF teker_8_j HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.2 -0.5 -0.25
      }
      device [
        RotationalMotor {
          name "motor8"
        }
      ]
      endPoint DEF teker_8 Solid {
        translation -0.2 -0.6 -0.249996
        rotation 1 1.1775671812919276e-16 -3.9252239376397585e-17 1.5707999999999998
        children [
          DEF teker_base Shape {
            appearance PBRAppearance {
              baseColor 0.505882 0.239216 0.611765
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.2
              radius 0.1
              subdivision 32
            }
          }
        ]
        name "teker_8"
        boundingObject USE teker_base
        physics Physics {
          density -1
          mass 1
        }
      }
    }
    DEF teker_7_j HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.15 -0.5 -0.25
      }
      device [
        RotationalMotor {
          name "motor7"
        }
      ]
      endPoint DEF teker_7 Solid {
        translation 0.15 -0.6 -0.249996
        rotation 1 7.850447875279517e-17 -3.9252239376397585e-17 1.5707999999999998
        children [
          DEF teker_base Shape {
            appearance PBRAppearance {
              baseColor 0.505882 0.239216 0.611765
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.2
              radius 0.1
              subdivision 32
            }
          }
        ]
        name "teker_7"
        boundingObject USE teker_base
        physics Physics {
          density -1
          mass 1
        }
      }
    }
    DEF teker_6_j HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.5 -0.5 -0.25
      }
      device [
        RotationalMotor {
          name "motor6"
        }
      ]
      endPoint DEF teker_6 Solid {
        translation 0.5 -0.6 -0.24999600000000002
        rotation 1 -3.9252239376397585e-17 0 1.5707999999999998
        children [
          DEF teker_base Shape {
            appearance PBRAppearance {
              baseColor 0.505882 0.239216 0.611765
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.2
              radius 0.1
              subdivision 32
            }
          }
        ]
        name "teker_6"
        boundingObject USE teker_base
        physics Physics {
          density -1
          mass 1
        }
      }
    }
    DEF teker_5_j HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.49 -0.5 -0.25
      }
      device [
        RotationalMotor {
          name "motor5"
        }
      ]
      endPoint DEF teker_5 Solid {
        translation -0.49 -0.6 -0.249996
        rotation 1 -5.887835906459636e-17 0 1.5707999999999998
        children [
          DEF teker_base Shape {
            appearance PBRAppearance {
              baseColor 0.505882 0.239216 0.611765
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.2
              radius 0.1
              subdivision 32
            }
          }
        ]
        name "teker_5"
        boundingObject USE teker_base
        physics Physics {
          density -1
          mass 1
        }
      }
    }
    DEF teker_4_j HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.2 0.5 -0.25
      }
      device [
        RotationalMotor {
          name "motor4"
        }
      ]
      endPoint DEF teker_4 Solid {
        translation -0.2 0.6 -0.249996
        rotation 1 2.1588731657018674e-16 -3.925223937639759e-17 1.5707999999999998
        children [
          DEF teker_base Shape {
            appearance PBRAppearance {
              baseColor 0.505882 0.239216 0.611765
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.2
              radius 0.1
              subdivision 32
            }
          }
        ]
        name "teker_4"
        boundingObject USE teker_base
        physics Physics {
          density -1
          mass 1
        }
      }
    }
    DEF teker_3_j HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.15 0.5 -0.25
      }
      device [
        RotationalMotor {
          name "motor3"
        }
      ]
      endPoint DEF teker_3 Solid {
        translation 0.15 0.6 -0.249996
        rotation 1 5.887835906459637e-17 0 1.5708
        children [
          DEF teker_base Shape {
            appearance PBRAppearance {
              baseColor 0.505882 0.239216 0.611765
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.2
              radius 0.1
              subdivision 32
            }
          }
        ]
        name "teker_3"
        boundingObject USE teker_base
        physics Physics {
          density -1
          mass 1
        }
      }
    }
    DEF teker_2_j HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.5 0.5 -0.25
      }
      device [
        RotationalMotor {
          name "motor2"
        }
      ]
      endPoint DEF teker_2 Solid {
        translation 0.5 0.6 -0.249996
        rotation 1 -1.3738283781739154e-16 1.962611968819879e-17 1.5708
        children [
          DEF teker_base Shape {
            appearance PBRAppearance {
              baseColor 0.505882 0.239216 0.611765
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.2
              radius 0.1
              subdivision 32
            }
          }
        ]
        name "teker_2"
        boundingObject USE teker_base
        physics Physics {
          density -1
          mass 1
        }
      }
    }
    DEF teker_1_j HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.49 0.5 -0.25
      }
      device [
        RotationalMotor {
          name "motor1"
        }
      ]
      endPoint DEF teker_1 Solid {
        translation -0.49 0.6 -0.24999599999999997
        rotation 1 -7.850447875279517e-17 0 1.5708
        children [
          DEF teker_base Shape {
            appearance PBRAppearance {
              baseColor 0.505882 0.239216 0.611765
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.2
              radius 0.1
              subdivision 32
            }
          }
        ]
        name "teker_1"
        boundingObject USE teker_base
        physics Physics {
          density -1
          mass 1
        }
      }
    }
    DEF Body Shape {
      appearance ThreadMetalPlate {
      }
      geometry Box {
        size 1 1 0.51
      }
    }
  ]
  name "skid_steer_car"
  boundingObject USE Body
  physics Physics {
    density -1
    mass 20
  }
  controller "skid_steer_kontrolcu"
}
